File: 1/sender/src/main/java/com/example/tutorials/one/Send.java
package com.example.tutorials.one;

import com.rabbitmq.client.ConnectionFactory;
import com.rabbitmq.client.Connection;
import com.rabbitmq.client.Channel;

public class Send {
    private final static String QUEUE_NAME = "hello";

    public static void main(String[] argv) throws Exception {

        // Establish connection to server
        ConnectionFactory factory = new ConnectionFactory();
        factory.setHost("localhost");
        Connection connection = factory.newConnection();
        Channel channel = connection.createChannel();

        // Bind to queue (create if not exists)
        // Queue: - non-durable (does not survive server restart)
        //        - non-exclusive (can be used by other connections)
        //        - non auto-delete (server doesn't delete when not in use)
        channel.queueDeclare(QUEUE_NAME, false, false, false, null);

        // Send message to queue
        String message = "Hello World!";
        channel.basicPublish("", QUEUE_NAME, null, message.getBytes());
        System.out.println(" [x] Sent '" + message + "'");

        // Close connection
        channel.close();
        connection.close();
    }
}

File: 1-hello-world/js/amqplib-promise-api/producer/send.js
#!/usr/bin/env node

// Establish connection to default RabbitMQ server on localhost, send one
// message to the queue, then exit.

var amqp = require('amqplib');
var queue = 'hello'
var msg = 'Hello World!'

amqp.connect().then(function(conn) {
  conn.createChannel().then(function(ch) {
    ch.assertQueue(queue, {durable: false});
    ch.sendToQueue(queue, Buffer.from(msg));
    console.log(" [x] Sent: %s", msg);
  });
  setTimeout(function() { conn.close(); process.exit(0) }, 500);
});
